File: binding/src/libqgraphicsgridlayout.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qgraphicsgridlayout.html
pub const qgraphicsgridlayout = struct {

    /// New constructs a new QGraphicsGridLayout object.
    pub fn New() QtC.QGraphicsGridLayout {
        return qtc.QGraphicsGridLayout_new();
    }


    /// New2 constructs a new QGraphicsGridLayout object.
    pub fn New2(parent: ?*anyopaque) QtC.QGraphicsGridLayout {
        return qtc.QGraphicsGridLayout_new2(@ptrCast(parent));
    }


    pub fn AddItem(self: ?*anyopaque, item: ?*anyopaque, row: i32, column: i32, rowSpan: i32, columnSpan: i32) void {
        qtc.QGraphicsGridLayout_AddItem(@ptrCast(self), @ptrCast(item), row, column, rowSpan, columnSpan);
    }

    pub fn AddItem2(self: ?*anyopaque, item: ?*anyopaque, row: i32, column: i32) void {
        qtc.QGraphicsGridLayout_AddItem2(@ptrCast(self), @ptrCast(item), row, column);
    }

    pub fn SetHorizontalSpacing(self: ?*anyopaque, spacing: f64) void {
        qtc.QGraphicsGridLayout_SetHorizontalSpacing(@ptrCast(self), spacing);
    }

    pub fn HorizontalSpacing(self: ?*anyopaque, ) f64 {
        return qtc.QGraphicsGridLayout_HorizontalSpacing(@ptrCast(self));
    }

    pub fn SetVerticalSpacing(self: ?*anyopaque, spacing: f64) void {
        qtc.QGraphicsGridLayout_SetVerticalSpacing(@ptrCast(self), spacing);
    }

    pub fn VerticalSpacing(self: ?*anyopaque, ) f64 {
        return qtc.QGraphicsGridLayout_VerticalSpacing(@ptrCast(self));
    }

    pub fn SetSpacing(self: ?*anyopaque, spacing: f64) void {
        qtc.QGraphicsGridLayout_SetSpacing(@ptrCast(self), spacing);
    }

    pub fn SetRowSpacing(self: ?*anyopaque, row: i32, spacing: f64) void {
        qtc.QGraphicsGridLayout_SetRowSpacing(@ptrCast(self), row, spacing);
    }

    pub fn RowSpacing(self: ?*anyopaque, row: i32) f64 {
        return qtc.QGraphicsGridLayout_RowSpacing(@ptrCast(self), row);
    }

    pub fn SetColumnSpacing(self: ?*anyopaque, column: i32, spacing: f64) void {
        qtc.QGraphicsGridLayout_SetColumnSpacing(@ptrCast(self), column, spacing);
    }

    pub fn ColumnSpacing(self: ?*anyopaque, column: i32) f64 {
        return qtc.QGraphicsGridLayout_ColumnSpacing(@ptrCast(self), column);
    }

    pub fn SetRowStretchFactor(self: ?*anyopaque, row: i32, stretch: i32) void {
        qtc.QGraphicsGridLayout_SetRowStretchFactor(@ptrCast(self), row, stretch);
    }

    pub fn RowStretchFactor(self: ?*anyopaque, row: i32) i32 {
        return qtc.QGraphicsGridLayout_RowStretchFactor(@ptrCast(self), row);
    }

    pub fn SetColumnStretchFactor(self: ?*anyopaque, column: i32, stretch: i32) void {
        qtc.QGraphicsGridLayout_SetColumnStretchFactor(@ptrCast(self), column, stretch);
    }

    pub fn ColumnStretchFactor(self: ?*anyopaque, column: i32) i32 {
        return qtc.QGraphicsGridLayout_ColumnStretchFactor(@ptrCast(self), column);
    }

    pub fn SetRowMinimumHeight(self: ?*anyopaque, row: i32, height: f64) void {
        qtc.QGraphicsGridLayout_SetRowMinimumHeight(@ptrCast(self), row, height);
    }

    pub fn RowMinimumHeight(self: ?*anyopaque, row: i32) f64 {
        return qtc.QGraphicsGridLayout_RowMinimumHeight(@ptrCast(self), row);
    }

    pub fn SetRowPreferredHeight(self: ?*anyopaque, row: i32, height: f64) void {
        qtc.QGraphicsGridLayout_SetRowPreferredHeight(@ptrCast(self), row, height);
    }

    pub fn RowPreferredHeight(self: ?*anyopaque, row: i32) f64 {
        return qtc.QGraphicsGridLayout_RowPreferredHeight(@ptrCast(self), row);
    }

    pub fn SetRowMaximumHeight(self: ?*anyopaque, row: i32, height: f64) void {
        qtc.QGraphicsGridLayout_SetRowMaximumHeight(@ptrCast(self), row, height);
    }

    pub fn RowMaximumHeight(self: ?*anyopaque, row: i32) f64 {
        return qtc.QGraphicsGridLayout_RowMaximumHeight(@ptrCast(self), row);
    }

    pub fn SetRowFixedHeight(self: ?*anyopaque, row: i32, height: f64) void {
        qtc.QGraphicsGridLayout_SetRowFixedHeight(@ptrCast(self), row, height);
    }

    pub fn SetColumnMinimumWidth(self: ?*anyopaque, column: i32, width: f64) void {
        qtc.QGraphicsGridLayout_SetColumnMinimumWidth(@ptrCast(self), column, width);
    }

    pub fn ColumnMinimumWidth(self: ?*anyopaque, column: i32) f64 {
        return qtc.QGraphicsGridLayout_ColumnMinimumWidth(@ptrCast(self), column);
    }

    pub fn SetColumnPreferredWidth(self: ?*anyopaque, column: i32, width: f64) void {
        qtc.QGraphicsGridLayout_SetColumnPreferredWidth(@ptrCast(self), column, width);
    }

    pub fn ColumnPreferredWidth(self: ?*anyopaque, column: i32) f64 {
        return qtc.QGraphicsGridLayout_ColumnPreferredWidth(@ptrCast(self), column);
    }

    pub fn SetColumnMaximumWidth(self: ?*anyopaque, column: i32, width: f64) void {
        qtc.QGraphicsGridLayout_SetColumnMaximumWidth(@ptrCast(self), column, width);
    }

    pub fn ColumnMaximumWidth(self: ?*anyopaque, column: i32) f64 {
        return qtc.QGraphicsGridLayout_ColumnMaximumWidth(@ptrCast(self), column);
    }

    pub fn SetColumnFixedWidth(self: ?*anyopaque, column: i32, width: f64) void {
        qtc.QGraphicsGridLayout_SetColumnFixedWidth(@ptrCast(self), column, width);
    }

    pub fn SetRowAlignment(self: ?*anyopaque, row: i32, alignment: i32) void {
        qtc.QGraphicsGridLayout_SetRowAlignment(@ptrCast(self), row, @intCast(alignment));
    }

    pub fn RowAlignment(self: ?*anyopaque, row: i32) i32 {
        return qtc.QGraphicsGridLayout_RowAlignment(@ptrCast(self), row);
    }

    pub fn SetColumnAlignment(self: ?*anyopaque, column: i32, alignment: i32) void {
        qtc.QGraphicsGridLayout_SetColumnAlignment(@ptrCast(self), column, @intCast(alignment));
    }

    pub fn ColumnAlignment(self: ?*anyopaque, column: i32) i32 {
        return qtc.QGraphicsGridLayout_ColumnAlignment(@ptrCast(self), column);
    }

    pub fn SetAlignment(self: ?*anyopaque, item: ?*anyopaque, alignment: i32) void {
        qtc.QGraphicsGridLayout_SetAlignment(@ptrCast(self), @ptrCast(item), @intCast(alignment));
    }

    pub fn Alignment(self: ?*anyopaque, item: ?*anyopaque) i32 {
        return qtc.QGraphicsGridLayout_Alignment(@ptrCast(self), @ptrCast(item));
    }

    pub fn RowCount(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsGridLayout_RowCount(@ptrCast(self));
    }

    pub fn ColumnCount(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsGridLayout_ColumnCount(@ptrCast(self));
    }

    pub fn ItemAt(self: ?*anyopaque, row: i32, column: i32) QtC.QGraphicsLayoutItem {
        return qtc.QGraphicsGridLayout_ItemAt(@ptrCast(self), row, column);
    }

    pub fn Count(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsGridLayout_Count(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnCount(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QGraphicsGridLayout_OnCount(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCount(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsGridLayout_QBaseCount(@ptrCast(self));
    }

    pub fn ItemAt2(self: ?*anyopaque, index: i32) QtC.QGraphicsLayoutItem {
        return qtc.QGraphicsGridLayout_ItemAt2(@ptrCast(self), index);
    }

    /// Allows for overriding the related default method
    pub fn OnItemAt2(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) QtC.QGraphicsLayoutItem) void {
        qtc.QGraphicsGridLayout_OnItemAt2(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseItemAt2(self: ?*anyopaque, index: i32) QtC.QGraphicsLayoutItem {
        return qtc.QGraphicsGridLayout_QBaseItemAt2(@ptrCast(self), index);
    }

    pub fn RemoveAt(self: ?*anyopaque, index: i32) void {
        qtc.QGraphicsGridLayout_RemoveAt(@ptrCast(self), index);
    }

    /// Allows for overriding the related default method
    pub fn OnRemoveAt(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QGraphicsGridLayout_OnRemoveAt(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseRemoveAt(self: ?*anyopaque, index: i32) void {
        qtc.QGraphicsGridLayout_QBaseRemoveAt(@ptrCast(self), index);
    }

    pub fn RemoveItem(self: ?*anyopaque, item: ?*anyopaque) void {
        qtc.QGraphicsGridLayout_RemoveItem(@ptrCast(self), @ptrCast(item));
    }

    pub fn Invalidate(self: ?*anyopaque, ) void {
        qtc.QGraphicsGridLayout_Invalidate(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnInvalidate(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsGridLayout_OnInvalidate(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInvalidate(self: ?*anyopaque, ) void {
        qtc.QGraphicsGridLayout_QBaseInvalidate(@ptrCast(self));
    }

    pub fn SetGeometry(self: ?*anyopaque, rect: ?*anyopaque) void {
        qtc.QGraphicsGridLayout_SetGeometry(@ptrCast(self), @ptrCast(rect));
    }

    /// Allows for overriding the related default method
    pub fn OnSetGeometry(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsGridLayout_OnSetGeometry(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetGeometry(self: ?*anyopaque, rect: ?*anyopaque) void {
        qtc.QGraphicsGridLayout_QBaseSetGeometry(@ptrCast(self), @ptrCast(rect));
    }

    pub fn SizeHint(self: ?*anyopaque, which: i32, constraint: ?*anyopaque) QtC.QSizeF {
        return qtc.QGraphicsGridLayout_SizeHint(@ptrCast(self), @intCast(which), @ptrCast(constraint));
    }

    /// Allows for overriding the related default method
    pub fn OnSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, ?*anyopaque) callconv(.c) QtC.QSizeF) void {
        qtc.QGraphicsGridLayout_OnSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSizeHint(self: ?*anyopaque, which: i32, constraint: ?*anyopaque) QtC.QSizeF {
        return qtc.QGraphicsGridLayout_QBaseSizeHint(@ptrCast(self), @intCast(which), @ptrCast(constraint));
    }

    pub fn AddItem6(self: ?*anyopaque, item: ?*anyopaque, row: i32, column: i32, rowSpan: i32, columnSpan: i32, alignment: i32) void {
        qtc.QGraphicsGridLayout_AddItem6(@ptrCast(self), @ptrCast(item), row, column, rowSpan, columnSpan, @intCast(alignment));
    }

    pub fn AddItem4(self: ?*anyopaque, item: ?*anyopaque, row: i32, column: i32, alignment: i32) void {
        qtc.QGraphicsGridLayout_AddItem4(@ptrCast(self), @ptrCast(item), row, column, @intCast(alignment));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QGraphicsGridLayout_Delete(@ptrCast(self));
    }
};

File: binding/src/libqgraphicsgridlayout.cpp
#include <QGraphicsGridLayout>
#include <QGraphicsLayout>
#include <QGraphicsLayoutItem>
#include <QRectF>
#include <QSizeF>
#include <qgraphicsgridlayout.h>
#include "libqgraphicsgridlayout.h"
#include "libqgraphicsgridlayout.hxx"

QGraphicsGridLayout* QGraphicsGridLayout_new() {
	 return new VirtualQGraphicsGridLayout();
}

QGraphicsGridLayout* QGraphicsGridLayout_new2(QGraphicsLayoutItem* parent) {
	 return new VirtualQGraphicsGridLayout(parent);
}

void QGraphicsGridLayout_AddItem(QGraphicsGridLayout* self, QGraphicsLayoutItem* item, int row, int column, int rowSpan, int columnSpan) {
	self->addItem(item, row, column, rowSpan, columnSpan);
}

void QGraphicsGridLayout_AddItem2(QGraphicsGridLayout* self, QGraphicsLayoutItem* item, int row, int column) {
	self->addItem(item, row, column);
}

void QGraphicsGridLayout_SetHorizontalSpacing(QGraphicsGridLayout* self, double spacing) {
	self->setHorizontalSpacing(spacing);
}

double QGraphicsGridLayout_HorizontalSpacing(const QGraphicsGridLayout* self) {
	return self->horizontalSpacing();
}

void QGraphicsGridLayout_SetVerticalSpacing(QGraphicsGridLayout* self, double spacing) {
	self->setVerticalSpacing(spacing);
}

double QGraphicsGridLayout_VerticalSpacing(const QGraphicsGridLayout* self) {
	return self->verticalSpacing();
}

void QGraphicsGridLayout_SetSpacing(QGraphicsGridLayout* self, double spacing) {
	self->setSpacing(spacing);
}

void QGraphicsGridLayout_SetRowSpacing(QGraphicsGridLayout* self, int row, double spacing) {
	self->setRowSpacing(row, spacing);
}

double QGraphicsGridLayout_RowSpacing(const QGraphicsGridLayout* self, int row) {
	return self->rowSpacing(row);
}

void QGraphicsGridLayout_SetColumnSpacing(QGraphicsGridLayout* self, int column, double spacing) {
	self->setColumnSpacing(column, spacing);
}

double QGraphicsGridLayout_ColumnSpacing(const QGraphicsGridLayout* self, int column) {
	return self->columnSpacing(column);
}

void QGraphicsGridLayout_SetRowStretchFactor(QGraphicsGridLayout* self, int row, int stretch) {
	self->setRowStretchFactor(row, stretch);
}

int QGraphicsGridLayout_RowStretchFactor(const QGraphicsGridLayout* self, int row) {
	return self->rowStretchFactor(row);
}

void QGraphicsGridLayout_SetColumnStretchFactor(QGraphicsGridLayout* self, int column, int stretch) {
	self->setColumnStretchFactor(column, stretch);
}

int QGraphicsGridLayout_ColumnStretchFactor(const QGraphicsGridLayout* self, int column) {
	return self->columnStretchFactor(column);
}

void QGraphicsGridLayout_SetRowMinimumHeight(QGraphicsGridLayout* self, int row, double height) {
	self->setRowMinimumHeight(row, height);
}

double QGraphicsGridLayout_RowMinimumHeight(const QGraphicsGridLayout* self, int row) {
	return self->rowMinimumHeight(row);
}

void QGraphicsGridLayout_SetRowPreferredHeight(QGraphicsGridLayout* self, int row, double height) {
	self->setRowPreferredHeight(row, height);
}

double QGraphicsGridLayout_RowPreferredHeight(const QGraphicsGridLayout* self, int row) {
	return self->rowPreferredHeight(row);
}

void QGraphicsGridLayout_SetRowMaximumHeight(QGraphicsGridLayout* self, int row, double height) {
	self->setRowMaximumHeight(row, height);
}

double QGraphicsGridLayout_RowMaximumHeight(const QGraphicsGridLayout* self, int row) {
	return self->rowMaximumHeight(row);
}

void QGraphicsGridLayout_SetRowFixedHeight(QGraphicsGridLayout* self, int row, double height) {
	self->setRowFixedHeight(row, height);
}

void QGraphicsGridLayout_SetColumnMinimumWidth(QGraphicsGridLayout* self, int column, double width) {
	self->setColumnMinimumWidth(column, width);
}

double QGraphicsGridLayout_ColumnMinimumWidth(const QGraphicsGridLayout* self, int column) {
	return self->columnMinimumWidth(column);
}

void QGraphicsGridLayout_SetColumnPreferredWidth(QGraphicsGridLayout* self, int column, double width) {
	self->setColumnPreferredWidth(column, width);
}

double QGraphicsGridLayout_ColumnPreferredWidth(const QGraphicsGridLayout* self, int column) {
	return self->columnPreferredWidth(column);
}

void QGraphicsGridLayout_SetColumnMaximumWidth(QGraphicsGridLayout* self, int column, double width) {
	self->setColumnMaximumWidth(column, width);
}

double QGraphicsGridLayout_ColumnMaximumWidth(const QGraphicsGridLayout* self, int column) {
	return self->columnMaximumWidth(column);
}

void QGraphicsGridLayout_SetColumnFixedWidth(QGraphicsGridLayout* self, int column, double width) {
	self->setColumnFixedWidth(column, width);
}

void QGraphicsGridLayout_SetRowAlignment(QGraphicsGridLayout* self, int row, int alignment) {
	self->setRowAlignment(row, static_cast<QFlags<Qt::AlignmentFlag>>(alignment));
}

int QGraphicsGridLayout_RowAlignment(const QGraphicsGridLayout* self, int row) {
	return self->rowAlignment(row);
}

void QGraphicsGridLayout_SetColumnAlignment(QGraphicsGridLayout* self, int column, int alignment) {
	self->setColumnAlignment(column, static_cast<QFlags<Qt::AlignmentFlag>>(alignment));
}

int QGraphicsGridLayout_ColumnAlignment(const QGraphicsGridLayout* self, int column) {
	return self->columnAlignment(column);
}

void QGraphicsGridLayout_SetAlignment(QGraphicsGridLayout* self, QGraphicsLayoutItem* item, int alignment) {
	self->setAlignment(item, static_cast<QFlags<Qt::AlignmentFlag>>(alignment));
}

int QGraphicsGridLayout_Alignment(const QGraphicsGridLayout* self, QGraphicsLayoutItem* item) {
	return self->alignment(item);
}

int QGraphicsGridLayout_RowCount(const QGraphicsGridLayout* self) {
	return self->rowCount();
}

int QGraphicsGridLayout_ColumnCount(const QGraphicsGridLayout* self) {
	return self->columnCount();
}

QGraphicsLayoutItem* QGraphicsGridLayout_ItemAt(const QGraphicsGridLayout* self, int row, int column) {
	return self->itemAt(row, column);
}

int QGraphicsGridLayout_Count(const QGraphicsGridLayout* self) {
	return self->count();
}

QGraphicsLayoutItem* QGraphicsGridLayout_ItemAt2(const QGraphicsGridLayout* self, int index) {
	return self->itemAt(index);
}

void QGraphicsGridLayout_RemoveAt(QGraphicsGridLayout* self, int index) {
	self->removeAt(index);
}

void QGraphicsGridLayout_RemoveItem(QGraphicsGridLayout* self, QGraphicsLayoutItem* item) {
	self->removeItem(item);
}

void QGraphicsGridLayout_Invalidate(QGraphicsGridLayout* self) {
	self->invalidate();
}

void QGraphicsGridLayout_SetGeometry(QGraphicsGridLayout* self, const QRectF* rect) {
	self->setGeometry(*rect);
}

QSizeF* QGraphicsGridLayout_SizeHint(const QGraphicsGridLayout* self, int which, const QSizeF* constraint) {
	return new QSizeF(self->sizeHint(static_cast<Qt::SizeHint>(which), *constraint));
}

void QGraphicsGridLayout_AddItem6(QGraphicsGridLayout* self, QGraphicsLayoutItem* item, int row, int column, int rowSpan, int columnSpan, int alignment) {
	self->addItem(item, row, column, rowSpan, columnSpan, static_cast<QFlags<Qt::AlignmentFlag>>(alignment));
}

void QGraphicsGridLayout_AddItem4(QGraphicsGridLayout* self, QGraphicsLayoutItem* item, int row, int column, int alignment) {
	self->addItem(item, row, column, static_cast<QFlags<Qt::AlignmentFlag>>(alignment));
}

// Base class handler implementation
int QGraphicsGridLayout_QBaseCount(const QGraphicsGridLayout* self) {
	auto* vqgraphicsgridlayout = dynamic_cast<const VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_Count_IsBase(true);
	return vqgraphicsgridlayout->count();
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsGridLayout_OnCount(const QGraphicsGridLayout* self, intptr_t slot) {
	auto* vqgraphicsgridlayout = dynamic_cast<const VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_Count_Callback(reinterpret_cast<VirtualQGraphicsGridLayout::QGraphicsGridLayout_Count_Callback>(slot));
}
}

// Base class handler implementation
QGraphicsLayoutItem* QGraphicsGridLayout_QBaseItemAt2(const QGraphicsGridLayout* self, int index) {
	auto* vqgraphicsgridlayout = dynamic_cast<const VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_ItemAt2_IsBase(true);
	return vqgraphicsgridlayout->itemAt(index);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsGridLayout_OnItemAt2(const QGraphicsGridLayout* self, intptr_t slot) {
	auto* vqgraphicsgridlayout = dynamic_cast<const VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_ItemAt2_Callback(reinterpret_cast<VirtualQGraphicsGridLayout::QGraphicsGridLayout_ItemAt2_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsGridLayout_QBaseRemoveAt(QGraphicsGridLayout* self, int index) {
	auto* vqgraphicsgridlayout = dynamic_cast<VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_RemoveAt_IsBase(true);
	vqgraphicsgridlayout->removeAt(index);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsGridLayout_OnRemoveAt(QGraphicsGridLayout* self, intptr_t slot) {
	auto* vqgraphicsgridlayout = dynamic_cast<VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_RemoveAt_Callback(reinterpret_cast<VirtualQGraphicsGridLayout::QGraphicsGridLayout_RemoveAt_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsGridLayout_QBaseInvalidate(QGraphicsGridLayout* self) {
	auto* vqgraphicsgridlayout = dynamic_cast<VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_Invalidate_IsBase(true);
	vqgraphicsgridlayout->invalidate();
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsGridLayout_OnInvalidate(QGraphicsGridLayout* self, intptr_t slot) {
	auto* vqgraphicsgridlayout = dynamic_cast<VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_Invalidate_Callback(reinterpret_cast<VirtualQGraphicsGridLayout::QGraphicsGridLayout_Invalidate_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsGridLayout_QBaseSetGeometry(QGraphicsGridLayout* self, const QRectF* rect) {
	auto* vqgraphicsgridlayout = dynamic_cast<VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_SetGeometry_IsBase(true);
	vqgraphicsgridlayout->setGeometry(*rect);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsGridLayout_OnSetGeometry(QGraphicsGridLayout* self, intptr_t slot) {
	auto* vqgraphicsgridlayout = dynamic_cast<VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_SetGeometry_Callback(reinterpret_cast<VirtualQGraphicsGridLayout::QGraphicsGridLayout_SetGeometry_Callback>(slot));
}
}

// Base class handler implementation
QSizeF* QGraphicsGridLayout_QBaseSizeHint(const QGraphicsGridLayout* self, int which, const QSizeF* constraint) {
	auto* vqgraphicsgridlayout = dynamic_cast<const VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_SizeHint_IsBase(true);
	return new QSizeF(vqgraphicsgridlayout->sizeHint(static_cast<Qt::SizeHint>(which), *constraint));
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsGridLayout_OnSizeHint(const QGraphicsGridLayout* self, intptr_t slot) {
	auto* vqgraphicsgridlayout = dynamic_cast<const VirtualQGraphicsGridLayout*>(self);
	if (vqgraphicsgridlayout && vqgraphicsgridlayout->isVirtualQGraphicsGridLayout) {
vqgraphicsgridlayout->setQGraphicsGridLayout_SizeHint_Callback(reinterpret_cast<VirtualQGraphicsGridLayout::QGraphicsGridLayout_SizeHint_Callback>(slot));
}
}

void QGraphicsGridLayout_Delete(QGraphicsGridLayout* self) {
    delete self;
}

